src/esp32/webhook.rs

```diff
@@ -0,0 +1,94 @@
+#[allow(unused_imports)] // need this trait for `get_attribute`
+use crate::common::config::Component;
+use crate::common::config::DynamicComponentConfig;
+use crate::proto::app::v1::ConfigResponse;
+use embedded_svc::{
+    http::{client::Client as HttpClient, Method},
+    io::Write,
+};
+use esp_idf_svc::http::client::{Configuration as HttpConfiguration, EspHttpConnection};
+use log::*;
+use serde_json::json;
+use thiserror::Error;
+
+#[derive(Debug, Error, PartialEq)]
+pub enum WebhookError {
+    #[error("`webhook` not present in board attributes")]
+    NoWebhook,
+    #[error("`webhook-secret` not present in board attributes")]
+    NoWebhookSecret,
+    #[error("Error with webhook request: {0}")]
+    RequestError(String),
+    #[error("Error in config: {0}")]
+    ConfigError(String),
+}
+
+/// Checks the board config for a `webhook` and `webhook-secret` attribute, sending a GET
+/// request with credentials for an SDK to make a connection.
+pub fn handle_webhook(config: ConfigResponse) -> Result<(), WebhookError> {
```

Comment:
I'd like to see this function split into two functions doing the following:
1) One that gets the webhook and secret from a `ConfigResponse`
2) One that sends the GET request to the webhook
I think doing this also means that the config could be dropped earlier

(Optional: I think having a struct that represents a Webhook and contains its secret and cloud fqdn would be nice. 1) above could be a constructor that takes in a `ConfigResponse` or `RobotConfig` instance and 2) would just be an instance method on the webhook)

Reply:
done